core-processor/src/ext.rs

```diff
@@ -662,6 +677,7 @@ impl EnvExt for Ext {
         let end = at
             .checked_add(len)
             .ok_or(MessageError::TooBigReadLen { at, len })?;
+        self.charge_gas_runtime_if_enough(RuntimeCosts::ReadPerByte(len))?;
```

Comment:
charging done after len checking to improve error handling?

Reply:
Yes, we will charge only if len is checked - made this just for fun